File: 22. Swap nodes in pairs.py
class Node:
    def __init__(self,data): self.data=data; self.next=None

def swap_pairs(head):
    dummy=Node(0); dummy.next=head
    prev=dummy
    while prev.next and prev.next.next:
        a=prev.next; b=a.next
        prev.next=b; a.next=b.next; b.next=a
        prev=a
    return dummy.next

def print_list(h):
    cur=h; out=[]
    while cur:
        out.append(str(cur.data)); cur=cur.next
    print(' -> '.join(out))

if __name__=='__main__':
    head=Node(1); head.next=Node(2); head.next.next=Node(3); head.next.next.next=Node(4)
    head=swap_pairs(head)
    print_list(head)

File: 22. Swap nodes in pairs.cpp
#include <iostream>
using namespace std; struct Node{int data; Node* next; Node(int d):data(d),next(nullptr){}
};
Node* swapPairs(Node* head){ if(!head||!head->next) return head; Node* newHead=head->next; Node* prev=nullptr; Node* cur=head; while(cur && cur->next){ Node* nxt=cur->next; Node* nextpair=nxt->next; nxt->next=cur; cur->next=nextpair; if(prev) prev->next=nxt; prev=cur; cur=nextpair; } return newHead; }
void print(Node* h){ while(h){ cout<<h->data; if(h->next) cout<<" -> "; h=h->next;} cout<<endl; }
int main(){ Node* head=new Node(1); head->next=new Node(2); head->next->next=new Node(3); head->next->next->next=new Node(4); head=swapPairs(head); print(head); return 0; }
